### DISCUSSION TITLE
Kubernetes installation

### DISCUSSION THREAD

**Discussion by KinableJan** [openintegrationhub/openintegrationhub#1466] [Q&A]
The documentation only covers local installation with minikube or installation on GCP.
To my suprise, there is no "general" k8s installation. 
It is not clear to me why GCP is bounded to this deployment. 
Doesn't this go against the principles of Kubernetes (cloud ignorant)?

I'm working on windows docker for desktop and was not able to install IOH after 4 hours of trial and error.
Some peaces are not even clear in the documentation, frases like "Afterwards, check if iam is existent on your cluster" without the how make it hard.

I'm not in a position to spend much time on this so it would be very helpful if someone could review this so it could work on any cloud platform.

Thanks!

**Comment by weberjm** [comment]
It is true that we provide only a couple of installation guides. The reason for this is that these guides cover the environments that our teams have used while building the framework. 

While the hope is to have K8s based applications be platform independent, in practice there are a lot of areas where the different providers have not standardised around one implementation. Specifically regarding our framework, which is composed of 20+ services with multiple communication methods between them, the Networking offerings of each service (and each local platform) differ to an extent that we are not able to develop and maintain guides for all of them internally. 

Additionally, security and custom deployment considerations mean that each cluster implementation can (and likely should) be built differently.
